### PR TITLE
Adds Trace Result screen 

### PR DESCRIPTION
**Pull request by eri9000**
### Description:
- Adds trace result screen with Function results

### Pre-merge Checklist

- a [vTest](https://runtime-kotlin.esri.com/view/all/job/vtest/job/toolkit/) Job for this PR has been run
  - [x] link: https://runtime-kotlin.esri.com/job/vtest/job/toolkit/98/
- Unit and/or integration tests have been added to exercise this PR's logic, and the tests are passing:
  - [ ] Yes
  - [ ] No

### CODE REVIEW

**Review by puneet-pdx — APPROVED** [review]
@eri9000 Changes look good. I guess we will plug up the values with the other issue in the sprint backlog 👍🏽

File: toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/ui/TraceNavigation.kt [hunk 1]
```diff
@@ -32,12 +34,16 @@ import com.arcgismaps.toolkit.utilitynetworks.TraceState
...
 @Composable
 internal fun TraceNavHost(navController: NavHostController, traceState: TraceState) {
+    val coroutineScope = rememberCoroutineScope()
     NavHost(navController = navController, startDestination = TraceNavRoute.TraceOptions.name) {
         composable(TraceNavRoute.TraceOptions.name) {
             TraceOptionsScreen(
                 configurations = emptyList(),
                 onPerformTraceButtonClicked = {
                     // TODO: Add call to perform trace
+                    coroutineScope.launch {
```
Comment (sorenoid) [inline toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/ui/TraceNavigation.kt:44]: can remove todo now.

File: toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/ui/TraceResultScreen.kt [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+                    modifier = Modifier.padding(end = 8.dp),
+                    text = traceName,
+                    style = MaterialTheme.typography.headlineMedium
+                )
+                Icon(
+                    imageVector = Icons.Outlined.MoreVert,
+                    contentDescription = ""
+                )
+            }
+            DropdownMenu(
```
Comment (sorenoid) [inline toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/ui/TraceResultScreen.kt:104]: Can you please match the style of the ColorPicker DropdownMenu? I think some rounding is better, but higher priority is that the menus are styled consistently.

File: toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/ui/TraceResultScreen.kt [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+    TraceResultScreen(traceResults, onClearAllResults = {}, onDeleteResult = {}, onZoomToResults = {})
+}
+/**
+ * Data class to hold the trace results.
+ *
+ * @param traceName The name of the trace.
+ * @param functionResults The list of function results.
+ * @since 200.6.0
+ */
+internal data class TraceResults(val traceName: String, val functionResults: List<FunctionResult>)
```
Comment (sorenoid) [inline toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/ui/TraceResultScreen.kt:219]: to aid smart recomposition

```suggestion
@Immutable
internal data class TraceResults(val traceName: String, val functionResults: List<FunctionResult>)
```

File: toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/ui/TraceResultScreen.kt [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+
+/**
+ * Data class to hold the function result.
+ *
+ * @param name The name of the function result.
+ * @param value The value of the function result.
+ * @param functionType The type of the function result.
+ * @since 200.6.0
+ */
+internal data class FunctionResult(val name: String, val value: String, val functionType: String)
```
Comment (sorenoid) [inline toolkit/utilitynetworks/src/main/java/com/arcgismaps/toolkit/utilitynetworks/ui/TraceResultScreen.kt:229]: to aid smart recomposition

```suggestion
@Immutable
internal data class FunctionResult(val name: String, val value: String, val functionType: String)
```

**Review by sorenoid — CHANGES_REQUESTED** [review]
looks great. I have a couple of suggestions

**Review by sorenoid — APPROVED** [review]

**Pull request merged**